### PR TITLE
classic cocktails

### PR DESCRIPTION
**Pull request by jppope**
**What does your Instant Answer do?**

Provides a list of frequently used cocktails

**What problem does your Instant Answer solve (Why is it better than organic links)?**

it will make if quicker to get multiple drink recipes for classic cocktails especially considering the organic results that normally display always make you dig through a whole bunch of ads to find a 4 line drink recipe

**What is the data source for your Instant Answer? (Provide a link if possible)**

Various, but the notable one => http://www.iba-world.com/index.php?option=com_content&view=article&id=88
I'm going to develop the list of classic cocktails to be a mashup of the technical "classics" decided by the IBA and other governing bodies 

**Why did you choose this data source?**

the IBA is a good list to start from. But generating a list of classic cocktails is an art form.  I've spent a decent time studying this concept from the grand masters. The bartenders of locations such as: Little Branch (NYC), AttaBoy (NYC), Apotheke (NYC), Franklin Mortgage (Philly), 320 main (orange county), Felix (Hong Kong), Absynthe (San Francisco), the Ranstead Room (Philly), Milk and honey (NYC), Please don't tell (NYC). So between the recommendations of those folks and a large (30 book library) of cocktail books... I'm just trying to put together a decent list.

The starting list is mostly "classics" with a couple of variations early to keep the list friendly 

**Are there any other alternative (better) data sources?**

not really.

**What are some example queries that trigger this Instant Answer?**

classic cocktails, cocktail recipes, manhattan recipe, How to make a margarita, etc

**Which communities will this Instant Answer be especially useful for? (gamers, book lovers, etc)**

Tech, foodies, and Cocktail junkies.

**Is this Instant Answer connected to a DuckDuckHack Instant Answer idea?**

no clue what that is

**Which existing Instant Answers will this one supercede/overlap with?**

nope.

**Are you having any problems? Do you need our help with anything?**

I'd like to break it down for instant answers that will give individual cocktails too

**Where did you hear about DuckDuckHack? (For first time contributors)**

meetup.com

**What does the Instant Answer look like? (Provide a screenshot for new or updated Instant Answers)**

thank you

------
IA Page: https://duck.co/ia/view/classic_cocktails_cheat_sheet

### CODE REVIEW

**Comment by MrChrisW** [comment]
Hi @jppope , thanks for your contribution. Do you mind filling out the [Goodie Pull Request Template](https://github.com/duckduckgo/zeroclickinfo-goodies/blob/master/pull_request_template_goodie.md) and adding it to your initial comment. This will provide the community a quick way to review your CheatSheet Goodie. Thanks :+1:

File: share/goodie/cheat_sheets/classic_cocktails.json [hunk 1]
```diff
@@ -0,0 +1,157 @@
+{  
+   "id":"Classic_Cocktails",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/classic_cocktails.json]: Please change to `classic_cocktails_cheat_sheet`

File: share/goodie/cheat_sheets/classic_cocktails.json [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+      "Brown Liquor",
+      "Summer",
+      "Old School",
+      "Brunch",
+      "Howard Hughes Style"
+   ],
+   "sections":{  
+      "Brown Liquor":[  
+         {  
+            "name":"Old Fashioned",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/classic_cocktails.json]: Whoops, there's a specific syntax you need to use in order for this to work.

Please take a look at the example in the CheatSheet documentation: https://duck.co/duckduckhack/goodie_cheat_sheets

The `sections` are made up of key-value pairs, which are used to build the UI you see here: https://duckduckgo.com/?q=tmux+cheat+sheet&ia=cheatsheet

**Comment by jppope** [comment]
Sorry about the syntax errors early ... didn't realize that it wouldn't just take pure json. I plan on expanding this list but I just wanted to get a base file started. Let me know if you need anything else. and one last note... "Unicorn Blood" is more of an underground classic between the best bartenders... it isn't classic in the "ubiquitous" sense that the rest are but its very much a classic in the sense that real bartenders know it and drink it, so i thought I would add it.

**Comment by jppope** [comment]
Just following up on this... I was really hoping to see it pop up :) how do we know about its inclusion?

File: share/goodie/cheat_sheets/classic_cocktails.json [hunk 3]
```diff
@@ -0,0 +1,144 @@
+{  
+   "id":"Classic_Cocktails_cheat_sheet",
```
Comment (zachthompson) [inline share/goodie/cheat_sheets/classic_cocktails.json]: All lower-case.

File: share/goodie/cheat_sheets/classic_cocktails.json [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+      "Brown Liquor",
+      "Summer",
+      "Old School",
+      "Brunch",
+      "Howard Hughes Style"
+   ],
+   "sections":{  
+      "Brown Liquor":[  
+         {  
+            "Old Fashioned",
```
Comment (zachthompson) [inline share/goodie/cheat_sheets/classic_cocktails.json]: "key":

File: share/goodie/cheat_sheets/classic_cocktails.json [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+            "Value":[
```
Comment (zachthompson) [inline share/goodie/cheat_sheets/classic_cocktails.json]: All "Value" keys need to be "val".

File: share/goodie/cheat_sheets/classic_cocktails.json [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+               "3 cl Gin",
+               "1.5 cl Fresh lemon juice",
+               "2 dashes Sugar syrup",
+               "6 cl Champagne"
+            ]
+         }
+      ],
+      "Howard Hughes Style":[  
+         {  
+            "Key":"Unicorn Blood",
```
Comment (zachthompson) [inline share/goodie/cheat_sheets/classic_cocktails.json]: Lower-case "k"

File: share/goodie/cheat_sheets/classic_cocktails.json [hunk 7]
```diff
@@ -0,0 +1,144 @@
...
+            "Value":[  
+               "4.5 cl Bourbon or Rye whiskey",
+               "2 Dashes Angostura Bitters",
+               "1 sugar cube",
+               "Few dashes plain water"
+            ]
+         },
+         {  
+            "key":"Manhattan",
+            "Value":[
```
Comment (zachthompson) [inline share/goodie/cheat_sheets/classic_cocktails.json]: Unfortunately we don't support displaying array items on separate lines.  So this won't look as expected.

**Comment by zachthompson** [comment]
After making all of the changes I mentioned in the line notes, this is what it will look like:

classic-cocktails

Without supporting newlines, I'm not sure this can work.

@jppope The file needs to be named `classic-cocktails.json` with a hypen instead of an underscore as well.

**Comment by moollaza** [comment]
@jppope sorry about the delay on this one! It somehow slipped past me! Really sorry.

My colleague @zachthompson has left some feedback outlining the changes that need to be made.

You'll be able to verify that the JSON actually works by testing your cheat sheet with DuckPAN. Whatever you see there is how it will look when it get's merged and is released live. We don't support arrays for the `val` and don't currently allow newline characters either. I think we'll have to opt for a comma-separate list for each `val`.

Also, we've now launched support for "templates" so if you add `"template_type": "reference"` this Cheat Sheet will look even more awesome!

File: share/goodie/cheat_sheets/classic_cocktails.json [hunk 8]
```diff
@@ -0,0 +1,144 @@
+{  
+   "id":"Classic_Cocktails_cheat_sheet",
+   "key":"Classic Cocktails",
+   "description":"The Top Classic Cocktails",
+   "metadata":{
```
Comment (moollaza) [inline share/goodie/cheat_sheets/classic_cocktails.json]: Let's add in the template specification here: `"template_type": "reference",`

**Comment by Razzeee** [comment]
This may also be nice as a spice, if you use http://www.thecocktaildb.com/

**Comment by moollaza** [comment]
> This may also be nice as a spice, if you use http://www.thecocktaildb.com/

@Razzeee thanks a *lot* for that link! We had a Spice until very recently that allowed us to search for drink making instructions, but the website suddenly went offline without notice. We'd love to replace the source and bring the IA back online. Would you be interested in doing it?

**Comment by Razzeee** [comment]
@moollaza 
I'm afraid I don't have the time required for that. As I would have to learn your technology stack fist and theres currently no time in my life for that. Sorry.

**Comment by moollaza** [comment]
>I'm afraid I don't have the time required for that. As I would have to learn your technology stack fist and >theres currently no time in my life for that. Sorry.

@Razzeee that's no problem, I'll create a task for the community to tackle this. Thanks again for sharing the API!

**Comment by moollaza** [comment]
@Razzeee FYI : https://github.com/duckduckgo/zeroclickinfo-spice/issues/2184 -- Would you mind sharing this on the Kodi forum?

**Comment by Razzeee** [comment]
Will do

**Comment by moollaza** [comment]
@jppope Ping! I'm just checking in to see if you're still interested in finishing this one up? We'd love to see this go live!

There's absolutely no rush, just wanted to make sure you're not stuck on anything.

For now I'll close this. If you want to pick it up again, we'll be sure to open it up for you.

**Comment by jppope** [comment]
Whoa...

Sorry... I thought we were good...

what all do I / did I need to do to get it live?

My apologies about the confusion, I thought when you were making those
changes that you were making them yourself and that I didn't have to go
back in and fix them.

**Pull request closed without merging**